pkg/server/options/embeddedetcd.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+)
+
+type EmbeddedEtcd struct {
+	Enable bool
+
+	Directory    string
+	PeerPort     string
+	ClientPort   string
+	WalSizeBytes int64
+}
+
+func NewEmbeddedEtcd() *EmbeddedEtcd {
+	return &EmbeddedEtcd{
+		Enable:     true,
+		Directory:  "",
+		PeerPort:   "2380",
+		ClientPort: "2379",
+	}
+}
+
+func (e *EmbeddedEtcd) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&e.Enable, "enable-embedded-etcd", e.Enable, "Enable embedded etcd. Will be ignored if --etcd-servers is specified")
```

Comment:
This name seems fine, although I'm a bit worried about adding flags that we don't directly and clearly need today (commented out or disabled here).  If WAL size is needed for integration I guess I can live with it.

Reply:
Have removed it. Now `--etcd-servers=embedded` is the default.

```
Start the control plane process

The server process listens on port 6443 and will act like a Kubernetes API server. It will initialize any necessary data to the provided start location or as a '.kcp' directory in the current directory. An admin kubeconfig file will be generated at initialization time that may be used to access the control plane.


Usage:
  kcp start [flags]

Most important flags:

      --etcd-servers strings
                List of etcd servers to connect with (scheme://ip:port), comma separated. By default an embedded etcd
                server is started. (default [embedded])
      --run-controllers
                Run the controllers in-process (default true)

Use "kcp start options" for a list of all flags available.
```